### PR TITLE
[Enhancement] accumulate chunk into full one in connector chunk source

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

ssb-100g with block cache, optimized version can save almost 1000ms. 

  | ssbflat(before) | after | ssb(before) | after
-- | -- | -- | -- | --
Q01 | 444 | 350 | 683 | 612
Q02 | 419 | 328 | 645 | 574
Q03 | 460 | 414 | 588 | 559
Q04 | 628 | 445 | 878 | 845
Q05 | 478 | 360 | 729 | 723
Q06 | 335 | 276 | 699 | 693
Q07 | 920 | 666 | 1372 | 1396
Q08 | 817 | 601 | 823 | 807
Q09 | 298 | 275 | 741 | 737
Q10 | 219 | 222 | 762 | 667
Q11 | 861 | 608 | 1573 | 1577
Q12 | 1027 | 748 | 1310 | 1168
Q13 | 958 | 738 | 1210 | 1095
SUM | 7864 | 6031 | 12013 | 11453

tpch-100g with block cache.

  | tpch(before) | after
-- | -- | --
Q01 | 3654 | 3655
Q02 | 513 | 478
Q03 | 2425 | 2293
Q04 | 1150 | 1069
Q05 | 1904 | 1874
Q06 | 896 | 774
Q07 | 1726 | 1682
Q08 | 1577 | 1530
Q09 | 4756 | 4699
Q10 | 2764 | 2590
Q11 | 380 | 379
Q12 | 1366 | 1204
Q13 | 2761 | 2796
Q14 | 1119 | 971
Q15 | 996 | 845
Q16 | 3535 | 3615
Q17 | 1869 | 1775
Q18 | 10364 | 10369
Q19 | 1173 | 1000
Q20 | 1220 | 1181
Q21 | 3513 | 3509
Q22 | 678 | 691
SUM | 50339 | 48979

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

**Review by silverbullet233 — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/exec/pipeline/scan/connector_scan_operator.cpp [hunk 1]
```diff
@@ -191,15 +192,31 @@ Status ConnectorChunkSource::_read_chunk(RuntimeState* state, vectorized::ChunkP
...
     }
 
-    do {
-        RETURN_IF_ERROR(_data_source->get_next(state, chunk));
-    } while ((*chunk)->num_rows() == 0);
+    while (_status.ok()) {
+        vectorized::ChunkPtr tmp;
+        _status = _data_source->get_next(state, &tmp);
+        if (_status.ok()) {
+            _ck_acc.push(std::move(tmp));
```
Comment (kangkaisen) [inline be/src/exec/pipeline/scan/connector_scan_operator.cpp:199]: if you use ChunkAccumulator in here, I think you need to some optimizations for More than half chunks like ChunkAccumulateOperator.  such as chunk num are: 4000, 3500, 4000, 3500

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/exec/pipeline/scan/connector_scan_operator.cpp [hunk 2]
```diff
@@ -191,15 +192,35 @@ Status ConnectorChunkSource::_read_chunk(RuntimeState* state, vectorized::ChunkP
...
-    return Status::OK();
+    if (_ck_acc.output_chunk() != nullptr || _ck_acc.staging_chunk() != nullptr) {
+        if (_ck_acc.output_chunk() != nullptr) {
+            *chunk = std::move(_ck_acc.output_chunk());
+        } else {
+            *chunk = std::move(_ck_acc.staging_chunk());
+        }
+        _rows_read += (*chunk)->num_rows();
+        return Status::OK();
+    }
```
Comment (ZiheLiu) [inline be/src/exec/pipeline/scan/connector_scan_operator.cpp:221]: How about also extracting `pull_chunk()`, `is_finished()` and `set_finished()`, etc. to `ChunkPipelineAccumulator` like follows? Then maybe we could hide the logic about `output_chunk` and `staging_chunk`, and needn't write some duplicated code.

```C++
while (_status.ok() && _ck_acc.need_input()) {
    _status = _data_source->get_next(state, &tmp);
    if (_status.ok()) {
        _ck_acc.push(std::move(tmp));
    } else if (!_status.is_end_of_file()) {
        return _status;
    } else {
        _ck_acc.set_finishing();
    }
}

// ...
if (_ck_acc.has_output()) {
    *chunk = _ck_acc.pull_chunk();
    return Status::OK();
}

_ck_acc.reset();
return Status::EndOfFile("");

```

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/exec/pipeline/scan/connector_scan_operator.cpp (hunk 2 above)
Comment (dirtysalt) [reply be/src/exec/pipeline/scan/connector_scan_operator.cpp]: Thanks. It's much more clear

**Comment by dirtysalt** [comment]
run starrocks_be_unittest

**Comment by dirtysalt** [comment]
run starrocks_clang-format

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by ZiheLiu — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**